test/library/packages/Sort/RadixSort/flagsortMSB.chpl

```diff
@@ -0,0 +1,142 @@
+
+module flagsortMSB {
+  use Sort;
+  use Time;
+  use BitOps;
+  use Random;
+  config param bucketBits=8;
+  param numBuckets = (1 << bucketBits);
+  param DISTRIBUTE_BUFFER = 1;
+  config const tryUnroll = false;
+
+   // To find a position where an element's count is to be updated in the counts array.
+   private proc findPosition(place:int, element:int, numBuckets:int) {
+     var position = ((element>>place) & (numBuckets-1));
+     return position;
+   }
+
+   // To find maximum element in an array
+   private proc findMaxElement(array:[] int) {
+     return max reduce array;
+   }
+
+   // To find the place value from which msb sorting starts
+   private proc findPlace(array:[] int) {
+
+     const maxEl = findMaxElement(array);    
+     var lz = clz(maxEl);
+     var numBits = 64 - lz ;
+     var place = ((numBits)>>3)<<3;
+
+     return place;
+   }
+  
+   // MSB initial call - helper function
+   proc radixSortMSB(array:[?dom] int) {
+
+     var place = findPlace(array);
+     __radixSortMSB(array,place,dom.low,dom.high);
+
+   }
+
+   private proc __radixSortMSB(array:[] int, place: int, startIndex: int, endIndex: int) {
+
+     if(endIndex - startIndex < 100) {
+       sort(array[startIndex..endIndex]);
```

Comment:
You need a return in this block.

The purpose of the block is to stop recursion, if you don't return, recursion won't stop ;)